pkg/utils/helm.go

```diff
@@ -35,3 +37,61 @@ func FilterHelmReleases(namespace string, releaseName string) ([]string, error)
 	}
 	return helmReleases, nil
 }
+
+func CheckK8sReady(namespace string) (bool, error) {
+	maxRetries := 10
+	retryInterval := 20 * time.Second
+
+	var pvcStatus, apiHealth bool
+	var err error
+
+	for i := 0; i < maxRetries; i++ {
+		pvcStatus, err = CheckPVCStatus(namespace)
+		if err != nil {
+			if i == maxRetries-1 {
+				return false, err
+			}
+			fmt.Printf("PVC status check failed (attempt %d/%d): %v\n", i+1, maxRetries, err)
+			time.Sleep(retryInterval)
+			continue
+		}
+
+		apiHealth, err = CheckK8sApiHealth(namespace)
+		if err != nil {
+			if i == maxRetries-1 {
+				return false, err
+			}
+			fmt.Printf("K8s API health check failed (attempt %d/%d): %v\n", i+1, maxRetries, err)
+			time.Sleep(retryInterval)
+			continue
+		}
+
+		if pvcStatus && apiHealth {
+			return true, nil
+		}
+
+		if i < maxRetries-1 {
+			fmt.Printf("K8s not ready yet. Retrying in %v... (attempt %d/%d)\n", retryInterval, i+1, maxRetries)
+			time.Sleep(retryInterval)
+		}
```

Comment:
Dont need this condition because we're in the for...loop condition

```suggestion
		fmt.Printf("K8s not ready yet. Retrying in %v... (attempt %d/%d)\n", retryInterval, i+1, maxRetries)
		time.Sleep(retryInterval)
```